ministryofjustice/laa-review-criminal-legal-aid
# Display mini reports by workstream on open/closed application pages

## Description of change
Changes the volumes report and workload report to filter by work stream. 
Note this is currently only changing the view on the open/closed applications pages to display the correct report data based on the work stream tab being selected. On the open applications page, this is the workload report. On the closed application page, this is the volume report. The report pages will be changed in a separate PR. 

As part of this PR, `work_stream` and `reviewed_on` are added to the review read model to set the work stream value on application received events and to scope the volume report data by the reviewed date 

## Link to relevant ticket
[CRIMAPP-231](https://dsdmoj.atlassian.net/browse/CRIMAPP-231)
[CRIMAPP-232](https://dsdmoj.atlassian.net/browse/CRIMAPP-232)

## Notes for reviewer
Must not be deployed without: https://github.com/ministryofjustice/laa-criminal-applications-datastore/pull/188#pullrequestreview-1757973222 which adds work stream to the application submission sns event

## Screenshots of changes (if applicable)

### Before changes:
The volume and workload reports on the open/closed application page currently shows data across all work streams
<img width="1772" alt="Screenshot 2023-11-30 at 17 46 57" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/c46cc166-34cc-4168-87b4-9f6d3f4aa2d0">
<img width="1772" alt="Screenshot 2023-11-30 at 17 46 48" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/3e4c15e9-7212-4f98-8c92-659281eab299">
<img width="1772" alt="Screenshot 2023-11-30 at 17 46 41" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/5d2995b9-dbda-4d20-951c-4d125549f834">
<img width="1772" alt="Screenshot 2023-11-30 at 17 46 34" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/b9dae716-242c-4a52-ae1f-694f3d37acfb">


### After changes:
The volume and workload reports on the open/closed application pages now show data by work stream (just showing extradition and CAT tabs for now as there are no NCT cases)

<img width="1772" alt="Screenshot 2023-11-30 at 17 39 13" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/41fc7636-7151-4bd3-8877-6c210c5f210c">
<img width="1772" alt="Screenshot 2023-11-30 at 17 39 05" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/2b48c8a2-da81-4a9a-bcad-3441e9b1c9ba">
<img width="1772" alt="Screenshot 2023-11-30 at 17 40 40" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/723b73a1-4840-4fb3-9b5a-8c7bcfe9d7d9">
<img width="1772" alt="Screenshot 2023-11-30 at 17 43 20" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/f911433f-6bfa-41af-bd6c-00c7af9f5f25">


## How to manually test the feature
Requires a extradition and CAT case to be open and an extradition and CAT case to be closed 

Navigate to the open applications page. Confirm that you see the right data mini report as per the selected work stream
Navigate to the closed applications page. Confirm that you see the right data mini report as per the selected work stream

Turn off the feature flag locally and confirm that you see all the application data in both mini reports 


[CRIMAPP-231]: https://dsdmoj.atlassian.net/browse/CRIMAPP-231?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CRIMAPP-232]: https://dsdmoj.atlassian.net/browse/CRIMAPP-232?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
